cadencefx/cadencefx.go

```diff
@@ -0,0 +1,81 @@
+// Copyright (c) 2017 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package cadencefx
+
+import (
+	"go.uber.org/fx"
+)
+
+// Module includes everything in ClientModule
+// Additionally, it provides constructor for:
+// 1. Cadence workers
+//
+// On start, it will spin up cadence workers based
+// on the specification in config/[base,development,prod].yaml
+//
+// Use Module if the FX app is a cadence worker.
+//
+// Sample yaml for configuring a worker:
+// cadence:
+//   workers:
+//	 - domain: domain1
+//	   task_list: tasklist1
+//     options:
+//        disableworkflowworker : true
+//   - domain: domain2
+//     task_list: tasklist2
+//
+var Module = fx.Options(
+	ClientModule,
+	fx.Provide(NewWorkers),
+	fx.Invoke(RegisterWorkers),
+)
+
+// ClientModule provides constructors for:
+// 1. Cadence thrift client
+// 2. Cadence client.Client
+// 3. Cadence domain client
```

Comment:
It seems godoc still [doesn't support lists](https://github.com/golang/go/issues/7873). So I think this needs to be indented so it doesn't get treated like a single paragraph.